### PR TITLE
For #18591 - Make the navbar sticky

### PR DESCRIPTION
**Pull request by Mugurell**
For #18591 

Needs https://github.com/mozilla-mobile/android-components/pull/10058

Result:

https://user-images.githubusercontent.com/11428869/114691545-773e0680-9d20-11eb-9967-3829a0736c60.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by mcarare — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:181]: We don't need to define stickiness here, since this is the old

**Review by eliserichards — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:181]: Other than that change, this is good to merge once the AC change is in 😄

**Review by eliserichards — APPROVED** [review]
Awesome! Once CI passes this is ready to land 🙌 💯

**Comment by eliserichards** [comment]
Unit test failure: 

```
e: /home/runner/work/fenix/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt: (163, 75): Cannot find a parameter with this name: isSticky
```

**Comment by Mugurell** [comment]
Having this locally while working on other things I see the app crashing somehow related to reportSiteIssuePlaceholder in the menu.
So this should not yet land, before this is fixed in AC.

**Comment by Mugurell** [comment]
Rebased on the current master which should include the needed AC code.

**Comment by Mugurell** [comment]
Seeing ` The test timed out. The test ran longer than its maximum allowed duration, and was stopped. `
Trying one more time.

**Comment by eliserichards** [comment]
Still timing out 🤔 I reached out in the release duty channel to see if anyone knows what's up with the timeouts

**Comment by eliserichards** [comment]
> If you look at [the videos](https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5684842378148438768/executions/bs.bb7b3a0f7de6d9f/videos), the test runner gets frozen when trying to open the three-dot menu. Since this PR is changes functionality in the menu itself, my guess is that the tests are failing for good reason, or need to be updated

**Comment by Mugurell** [comment]
Went through all failed tests, ran them locally just fine, updated just a few which seemed to be impacted by this change.
Hopefully the above error affecting most of them will have been just a fluke.

**Comment by Mugurell** [comment]
Adding @sv-ohorvath as a reviewer for the ui tests part.

**Review by sv-ohorvath — APPROVED** [review]
The changes to UI tests look ok to me.

**Comment by eliserichards** [comment]
Looks like one last failure. 
`openMainMenuAddToCollectionTest `: "androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (with res-name that is "name_collection_edittext")"

**Comment by Mugurell** [comment]
One unit test failing now
```
TEST: onBookmarkTabs calls navigation on DefaultNavigationInteractor
TEST: test shouldDisplay
    [Robolectric] org.mozilla.fenix.components.tips.providers.MigrationTipProviderTest.test shouldDisplay: sdk=28; resources=BINARY
    I/MonitoringInstr: Instrumentation started!
FAILURE

java.lang.AssertionError: Verification failed: call 2 of 2: AddBookmarksUseCase(child of #506#510).invoke(any(), any(), any(), any())) was not called
````

Seems to be the one tracked in https://github.com/mozilla-mobile/fenix/issues/19024. Restarting CI tasks.

**Pull request merged**
